app-misc/freeplane/freeplane-1.6.15.ebuild

```diff
@@ -0,0 +1,56 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+# based on: freeplane-1.5.18.ebuild
+
+EAPI=7
+
+inherit virtualx
+
+DESCRIPTION="Java application for Mind Mapping, Knowledge and Project Management"
+HOMEPAGE="https://www.freeplane.org/"
+SRC_URI="mirror://sourceforge/${PN}/${PN}%20stable/${PN}_src-${PV}.tar.gz
+	https://github.com/rtgiskard/pkgs/raw/master/gentoo/distfiles/${PF}-gradle-cache.tar.xz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+
+JAVA_PKG_IUSE="doc"
+VIRTUALX_REQUIRED="test"
+
+RDEPEND=">=virtual/jre-1.8"
+DEPEND="${DEPEND}
+	>=virtual/jdk-1.8
+	dev-java/gradle-bin"
+
+src_compile() {
+	GRADLE="gradle --gradle-user-home ${WORKDIR}/.gradle --console rich --no-daemon"
+	GRADLE="${GRADLE} --offline"
+
+	${GRADLE} build -x check -x test || die
+}
+
+src_test() {
+	# disable --offline and then generate cache file with: ebuild clean test
+	echo "pack cache file to reuse for offline building .."
+	tar --xz -cf "${WORKDIR}/${PF}-gradle-cache.tar.xz" -C "${WORKDIR}" .gradle/caches/modules-2
```

Comment:
This needs `|| die` as well, although I think there's a portage built-in command to achieve unpacking (not sure though!)